packages/components/src/templates/next/components/internal/SearchableTable/getFilteredItems.ts

```diff
@@ -0,0 +1,14 @@
+import type { SearchableTableClientProps } from "./SearchableTableClient"
+
+interface GetFilteredItemsParams {
+  items: SearchableTableClientProps["items"]
+  searchValue: string
+}
+
+export const getFilteredItems = ({
+  items,
+  searchValue,
+}: GetFilteredItemsParams) =>
+  items
+    .filter((item) => item.key.includes(searchValue.toLowerCase()))
```

Comment:
if we want a case insensitive search, might as well lowercase the key!

Reply:
Hmm the key is lowercased in the server component so the client does lesser stuff: https://github.com/opengovsg/isomer/pull/713/files#diff-5c209342b9152356c25bfdfc004d41eeb44c795172426b1bc17569d3b0e31bdb

Reply:
not very good to assume here, we should just lowercase if it's cheap to do so. i think the runtime here won't be anyting significant

Reply:
Hmm this is done as part of a performance improvement suggested here: https://github.com/opengovsg/isomer/pull/713#discussion_r1782061929

I'm more inclined to reduce the work that the client does, especially since our users may have devices that are extremely slow.

Reply:
ok! can add a comment on the studio side then, so people are aware of this!

Reply:
Eh hmm there's no studio equivalent here, so the input is fine but on build time the key is generated with the lowercase.